tests/integration/test_sns.py

```diff
@@ -776,6 +789,51 @@ def receive_dlq():
 
         retry(receive_dlq, retries=10, sleep=2)
 
+    def test_dlq_receives_multiple_messages(
+        self, sqs_client, sns_client, sqs_create_queue, sns_create_topic, sns_subscription
+    ):
+        dlq_name = f"dlq-{short_uid()}"
+        dlq_url = sqs_create_queue(QueueName=dlq_name)
+        dlq_arn = get_queue_arn(sqs_client, dlq_url)
+
+        target_queue_name = f"target-{short_uid()}"
+        target_queue_url = sqs_create_queue(QueueName=target_queue_name)
+        target_target_arn = get_queue_arn(sqs_client, target_queue_url)
+
+        sns_topic_arn = sns_create_topic()["TopicArn"]
+        subscription = sns_subscription(
+            TopicArn=sns_topic_arn,
+            Protocol="sqs",
+            Endpoint=target_target_arn,
+        )
+
+        sns_client.set_subscription_attributes(
+            SubscriptionArn=subscription["SubscriptionArn"],
+            AttributeName="RedrivePolicy",
+            AttributeValue=json.dumps({"deadLetterTargetArn": dlq_arn}),
+        )
+        sns_client.publish(
+            TopicArn=sns_topic_arn, Message=json.dumps({"message": "target_message"})
+        )
+
+        def receive_message(url, msg, is_msg_list=False):
+            result = sqs_client.receive_message(QueueUrl=url, MessageAttributeNames=["All"])
+            assert len(result["Messages"]) > 0
+            msg_received = json.loads(result["Messages"][0]["Body"])["Message"]
+            if is_msg_list:
+                msg_received = msg_received[0]
+            msg_received = json.loads(msg_received)["message"]
+            assert msg == msg_received
+
+        retry(partial(receive_message, target_queue_url, "target_message"), retries=10, sleep=2)
```

Comment:
nit: while i find the approach using `partial` quite clever, this is not a pattern we typically use. i think it would be better to just call it using a lambda
```suggestion
        retry(lambda: receive_message(target_queue_url, "target_message"), retries=10, sleep=2)
```